### DISCUSSION TITLE
Stairs?

### DISCUSSION THREAD

**Discussion by dangerousthing** [nasa-jpl/open-source-rover#437] [General]
Will a rover built to specifications handle the stairs of a normal US apartment?

**Comment by Achllle** [comment]
As I wrote in [this comment](https://github.com/nasa-jpl/open-source-rover/discussions/234#discussioncomment-7541621) to you, it depends. There is no 'normal US apartment' when it comes to stairs. The closest I've seen is [California's code for stairs](https://www.dir.ca.gov/title8/3231.html) which recommends a range of angles, but it really depends on what the tread profile looks like, like the 'rise' of the tread, whether there is a 'nosing', and the material (carpet, wood, ...). The presence of nosing or incline on the stairs, especially in addition with a slippery surface, makes it hard for the wheels to grip the surface and climb up. The spacing between treads is important such that the wheels don't get stuck in the corners at the same time.

In the 'general' case, without any software modifications, I don't think you'll find that the rover will be able to easily drive up without problem.

As a side project, I'm building a stairlift specifically designed for robots. If that sounds interesting to you, you're welcome to reach out via e.g. Slack.

**Comment by ericjunkins** [comment]
@dangerousthing as Achille said it does come down to the specific dimensions of the stairs. One single stair at a time is no problem, however the issue comes if the front wheels are trying to climb up one stair while the back two are still trying to get up the previous step, it mostly needs to have 2 sets of wheels with good traction at any point to obstacle climb
